x-pack/plugins/lists/public/exceptions/components/builder/exception_items_renderer.tsx

```diff
@@ -131,6 +132,9 @@ export const ExceptionBuilderComponent = ({
     disableNested: isNestedDisabled,
     disableOr: isOrDisabled,
   });
+  const [currentExceptionListItems, setCurrentExceptionListItems] = useState<
```

Comment:
maybe using `useRef` would be more preferable here as it would not trigger another re-render

```ts
  const exceptionsRef = useRef<ExceptionsBuilderExceptionItem[]>([]);

  useEffect(() => {
    if (exceptionListItems.length > 0 && !deepEqual(exceptionListItems, exceptionsRef.current)) {
      exceptionsRef.current = exceptionListItems;
      setUpdateExceptions(exceptionListItems);
    }
  }, [exceptionListItems, setUpdateExceptions]);

```

Reply:
I tried it but it didn't work, and I am not sure if it will as the previous reason :( 

> Upon further investigation, I discovered that modifying the entries of the Exception and subsequently changing another field causes a reset to the initial entries. This behavior occurs because altering the entries (conditions) does not trigger a corresponding change in the exceptionListItems.